### PR TITLE
[TD-11078]<fix>: fix backend plugin error in case of integer types

### PR DESCRIPTION
**Pull request by zitsen**

### CODE REVIEW

File: pkg/tdengine.go [hunk 1]
```diff
@@ -107,54 +107,63 @@ func generateSql(query backend.DataQuery) (sql, alias string, err error) {
...
+		// BOOL
 		return []bool{}
-	} else if typeNum == 2 { // 2：TINYINT
-		return []int8{}
-	} else if typeNum == 3 { // 3：SMALLINT
-		return []int16{}
-	} else if typeNum == 4 { // 4：INT
-		return []int32{}
-	} else if typeNum == 5 { // 5：BIGINT
+	} else if (typeNum >= 2 && typeNum <= 5) || (typeNum >= 11 && typeNum <= 14) {
```
Comment (sangshuduo) [inline pkg/tdengine.go:113]: so many magic number..

Comment (zitsen) [reply pkg/tdengine.go]: Fixed

**Review by afwerar — APPROVED** [review]
LGTM

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**